src/policy/policy.cpp

```diff
@@ -266,8 +300,17 @@ bool IsWitnessStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
             // Taproot spend (non-P2SH-wrapped, version 1, witness program size 32; see BIP 341)
             Span stack{tx.vin[i].scriptWitness.stack};
             if (stack.size() >= 2 && !stack.back().empty() && stack.back()[0] == ANNEX_TAG) {
-                // Annexes are nonstandard as long as no semantics are defined for them.
-                return false;
+                // An annex is present. Remove it from the stack and save it for
+                // standardness checks.
+                const auto& annex = SpanPopBack(stack);
+
+                // Check that the annex is standard.
+                if (!IsAnnexStandard(annex)) {
```

Comment:
Which DoS concerns are you referring to? Since #24007 hasn't landed yet, the tx can only be accepted once?

Reply:
See the CPU DoS concerned mentioned above about BIP341’s `sha_annex`, if annexes start to be relayed.

Reply:
Yes, that's clear now. Thanks.

Same as with `MAX_ANNEX_BUDGET`, my feeling is that this safeguard isn't necessary because the extra cost is only linear and paid for.